### PR TITLE
Merge flash swap contracts

### PR DESCRIPTION
**Pull request by paulrberg**
See commits for changelog.

### CODE REVIEW

File: packages/flash-swap/test/shared/types.ts [hunk 1]
```diff
@@ -38,5 +38,4 @@ export interface Signers {
   borrower: SignerWithAddress;
   liquidator: SignerWithAddress;
   raider: SignerWithAddress;
-  subsidizer: SignerWithAddress;
```
Comment (scorpion9979) [inline packages/flash-swap/test/shared/types.ts:41]: good catch

File: packages/flash-swap/test/shared/fixtures.ts [hunk 2]
```diff
@@ -13,20 +13,18 @@ import { GodModeUniswapV2Pair__factory } from "../../src/types/factories/GodMode
...
 import type { MaliciousPair } from "../../src/types/MaliciousPair";
 import type { SimplePriceFeed } from "../../src/types/SimplePriceFeed";
 import type { GodModeUniswapV2Pair } from "../../src/types/GodModeUniswapV2Pair";
 import { deployGodModeErc20 } from "./deployers";
 
 type IntegrationFixtureReturnType = {
   balanceSheet: BalanceSheetV1;
-  collateralFlashUniswapV2: CollateralFlashUniswapV2;
+  flashUniswapV2: FlashUniswapV2;
   fintroller: FintrollerV1;
```
Comment (scorpion9979) [inline packages/flash-swap/test/shared/fixtures.ts:25]: `fintroller` needs to be defined above `flashUniswapV2`

File: packages/flash-swap/contracts/uniswap-v2/FlashUniswapV2.sol [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                    (collateralReserves, underlyingReserves, ) = pair.getReserves();
+                }
+
+                numerator = collateralReserves * underlyingAmount * 1000;
+                denominator = (underlyingReserves - underlyingAmount) * 997;
+            } else {
+                numerator = underlyingAmount * 1000;
+                denominator = 997;
+            }
+            repayAmount = numerator / denominator + 1;
```
Comment (scorpion9979) [inline packages/flash-swap/contracts/uniswap-v2/FlashUniswapV2.sol:82]: @paulrberg not sure why we're not using unchecked math here

File: packages/flash-swap/contracts/uniswap-v2/FlashUniswapV2.sol [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        );
+
+        // Note that "turnout" is a signed int. When it is negative, it acts as a maximum subsidy amount.
+        // When its value is positive, it acts as a minimum profit.
+        if (int256(vars.seizeAmount) < int256(vars.repayAmount) + vars.turnout) {
+            revert FlashUniswapV2__TurnoutNotSatisfied(vars.seizeAmount, vars.repayAmount, vars.turnout);
+        }
+
+        // Transfer the subsidy amount.
+
```
Comment (scorpion9979) [inline packages/flash-swap/contracts/uniswap-v2/FlashUniswapV2.sol:151]: let's remove the newline here

Comment (paulrberg) [reply packages/flash-swap/contracts/uniswap-v2/FlashUniswapV2.sol:151]: Good catch!

**Comment by paulrberg** [comment]
I added a new commit. @scorpion9979 - can you review?

**Pull request merged**